DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,419 @@
+            //     if (modifiedDay - 1 == 0) {
+            //         if (modifiedMonth - 1 < 1) {
+            //             modifiedYear--;
+            //             modifiedMonth = monthsInYear;
+            //         } else {
+            //             modifiedMonth--;
+            //         }
+            //         currentDaysInMonth = isLeapYear(modifiedYear) && modifiedMonth == 2 ? Date.Month.values()[modifiedMonth - 1].days + 1 : Date.Month.values()[modifiedMonth - 1].days;   
+            //         modifiedDay = currentDaysInMonth; 
+            //     } else {
+            //         modifiedDay--;
+            //     }
+            //     days--;
+            // }
+
+            // modifiedDay -= days;
+            return new Date(modifiedYear, Date.Month.values()[modifiedMonth - 1], modifiedDay);
+        } else {
+            throw new IllegalArgumentException("Invalid integer entered");
+        }
+        // return new Date(modifiedYear, Date.Month.values()[modifiedMonth - 1], modifiedDay);
+
+        // int y = this.year;
+        // int m = this.month.ordinal() + 1;
+        // int d = this.day;
+
+
+        // if (years >= 0 && months >= 0 && days >= 0) {
+        //     y -= years;
+            
+        //     if (m - months < 1){
+        //         while (m - months < 1){
+        //             y--;
+        //             months -= 12;
+        //         }
+        //     }
+        //     m -= months;
+
+        //     if (d - days < 1){
+        //         while(d - days < 1) {
+        //             if (m - 1 < 1){
+        //                 y--;
+        //                 m = 12;
+        //             } else {
+        //                 m--;
+        //             }
+
+        //             if (m == 2) {
+        //                 if (isLeapYear(y)) {
+        //                     days -= Date.Month.values()[m - 1].days + 1;
+        //                 }
+        //             } 
+
+        //             days -= Date.Month.values()[m - 1].days;
+        //         }
+        //     } 
+        //     d -= days;
+        // } else {
+        //     throw new IllegalArgumentException("Invalid input");
+        // }
+        
+        //Returning new object is a design decision. Unsure if new object or reference would be needed, to play it safe returning new obj. Example: returning vacation days might need to be a new hash.
+        // return new Date(y, Date.Month.values()[m - 1], d);
+    }
+
+    public int diff(Date input){
+        //return in value of days
+        int payload = 0;
+        int currentDays = (int)convToDays(this.year, this.month.ordinal() + 1, this.day);
+        int inputDays = (int)convToDays(input.year, input.month.ordinal() + 1, input.day);
+
+        if (currentDays > inputDays){
+            payload = currentDays - inputDays;
+        } else {
+            payload = inputDays - currentDays;
+        }
```

Comment:
Suggestion, use [Math.abs](https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#abs(int)):
```java
payload = Math.abs(currentDays - inputDays);
```
That way you will get always the difference in positive

Reply:
Excellent, I was hoping there would be a method for this that already existed. Also on a side/funny note, it took me way to long to realize it's not abs as abdominal but for absolute. To many jokes about math having a strong core.